ni_measurementlink_service/session_management.py

```diff
@@ -138,6 +138,20 @@ def unreserve(self):
         """Unreserve sessions."""
         self._session_manager._unreserve_sessions(self._session_info)
 
+    def get_pin_based_session_info(self, pin: str) -> List[Tuple[SessionInformation, str]]:
```

Comment:
I suspect there should be two versions of this:
- Get a single session/channel by pin and optional site. If this matches multiple sessions, throw an exception. Return a single session/channel.
- Get multiple sessions/channels by pin and optional site. Return all of the session/channel pairs that match.

Reply:
The reason I did not have an API which accepts the site information, was because in case of measurement link the pin context always is based on the site selection in instrument studio and usually the site selection is a single site. Only when the setting is `All Sites` multiple sites might come into picture. Even in the multiple site scenario, I assume the measurement operations are going to be repeated and not unique for each of the site. This was the primary reason i felt that the measurement author might not require to provide the site information as this logic might restrict their options in the instrument studio settings.
If we wanted to go with two helper APIs can we have something like this,
1. A method which accepts only pin named 'get_session_info_for_pin` which only accepts pin name and returns an array of session information.
2. A method which accepts both pin and site `get_session_info_for_pin_and_site` which accepts both pin and site and returns a array of session information.

Reply:
"optional site" is orthogonal to "get single session" vs. "get multiple sessions". If you remove "optional site" from my suggestion, the rest still stands.

If "optional site" is a less important use case, then it would be more appropriate to support it via an optional parameter than via a separate function.

```
if pin == channel_mapping.pin_or_relay_name and (site is None or site == channel_mapping.site):
```

Reply:
I have updated the PR and taken the approach of having a single method with site as optional input